src/sancus_support/sm_support.h

```diff
@@ -328,7 +328,8 @@ always_inline int is_buffer_outside_region(void *start, void *end,
     buf_end = buf;
   }
 
-  if( (end <= buf) || (start > buf_end) ) {
+  /* check for int overflow and finally validate `buf` falls outside */ 
+  if( (buf <= buf_end) && (end <= buf) || (start > buf_end) ) {
```

Comment:
@jovanbulck I think we need brackets around the second term of the AND operator:
```c
if( (buf <= buf_end) && ((end <= buf) || (start > buf_end)) ) { ... }
```

Reply:
yes, you're completely right, good catch! :sweat_smile:

Reply:
applied the fix, thanks @gianlu33 for the catch